src/LndClient/Data/LndEnv.hs

```diff
@@ -62,69 +65,119 @@ data LndEnv
         envLndLogErrors :: Bool
       }
 
+instance ToGrpc LndWalletPassword ByteString where
+  toGrpc x = Right $ encodeUtf8 (coerce x :: Text)
+
+instance Read LndTlsCert where
+  readsPrec _ x =
+    case createLndTlsCert $ C8.pack x of
+      Right r -> [(r, "")]
+      Left _ -> case reads x of
+        [(r, "")] -> [(r, "")]
+        _ -> []
+
+instance FromJSON LndTlsCert where
+  parseJSON x =
+    case x of
+      A.String s ->
+        case createLndTlsCert $ C8.pack $ T.unpack s of
+          Right cert -> return cert
+          Left e -> failure e
+      e -> failure e
```

Comment:
It's nice to have some tests where `Read` and `FromJSON` instances are failing, just to check that our implementation is correct and smart constructor is applied. Not big deal, might be separate pull request.